ocs_ci/deployment/cnv.py

```diff
@@ -0,0 +1,505 @@
+            namespace_yaml = OCS(**namespace_yaml_file)
+            namespace_yaml.create()
+            logger.info(f"CNV namespace {self.namespace} was created successfully")
+        except exceptions.CommandFailed as ef:
+            if (
+                f'project.project.openshift.io "{self.namespace}" already exists'
+                not in str(ef)
+            ):
+                logger.info(f"Namespace {self.namespace} already present")
+                raise ef
+
+    def create_cnv_operatorgroup(self):
+        """
+        Creates an OperatorGroup for CNV
+
+        """
+        operatorgroup_yaml_file = templating.load_yaml(CNV_OPERATORGROUP_YAML)
+        operatorgroup_yaml = OCS(**operatorgroup_yaml_file)
+        operatorgroup_yaml.create()
+        logger.info("CNV OperatorGroup created successfully")
+
+    def create_cnv_subscription(self):
+        """
+        Creates subscription for CNV operator
+
+        """
+        # Create an operator group for CNV
+        logger.info("Creating OperatorGroup for CNV")
+        self.create_cnv_operatorgroup()
+        cnv_subscription_yaml_file = CNV_SUBSCRIPTION_YAML
+        cnv_subscription_yaml_data = templating.load_yaml(cnv_subscription_yaml_file)
+        cnv_channel_version = config.DEPLOYMENT.get("ocs_csv_channel")[-4:]
+        cnv_sub_channel = f"nightly-{cnv_channel_version}"
+        cnv_subscription_yaml_data["spec"]["channel"] = f"{cnv_sub_channel}"
+        cnv_subscription_manifest = tempfile.NamedTemporaryFile(
+            mode="w+", prefix="cnv_subscription_manifest", delete=False
+        )
+        templating.dump_data_to_temp_yaml(
+            cnv_subscription_yaml_data, cnv_subscription_manifest.name
+        )
+        logger.info("Creating subscription for CNV operator")
+        run_cmd(f"oc create -f {cnv_subscription_manifest.name}")
+        self.wait_for_the_resource_to_discover(
+            kind=constants.SUBSCRIPTION,
+            namespace=self.namespace,
+            resource_name=constants.KUBEVIRT_HYPERCONVERGED,
+        )
+        wait_for_install_plan_and_approve(self.namespace)
+        cnv_package_manifest = PackageManifest(
+            resource_name=constants.KUBEVIRT_HYPERCONVERGED,
+            subscription_plan_approval="Manual",
+            install_plan_namespace=self.namespace,
+        )
+        # Wait for package manifest is ready
+        cnv_package_manifest.wait_for_resource(
+            resource_name=constants.KUBEVIRT_HYPERCONVERGED, timeout=300
+        )
+        csv = get_csvs_start_with_prefix(
+            csv_prefix=constants.KUBEVIRT_HCO_PREFIX, namespace=self.namespace
+        )
+        csv_name = csv[0]["metadata"]["name"]
+        csv_obj = CSV(resource_name=csv_name, namespace=self.namespace)
+        csv_obj.wait_for_phase(phase="Succeeded", timeout=720)
+
+    def wait_for_the_resource_to_discover(self, kind, namespace, resource_name):
```

Comment:
if this is a generic function please move this to utils or any suitable common utility function modules